Fixed in hg around 5354:d4ea7f571f10; remove as soon as there's no PATH_MAX in the code anymore.

--- a/mutt.h
+++ b/mutt.h
@@ -52,6 +52,11 @@
 #include <limits.h>
 #endif
 
+/* PATH_MAX is undefined on the hurd */
+#ifndef PATH_MAX
+#define PATH_MAX _POSIX_PATH_MAX
+#endif
+
 #include <pwd.h>
 #include <grp.h>
 
